### PR TITLE
Remove unused headers in several package.

### PR DESCRIPTION
**Pull request by yuanchao**
#### PR description:

Remove unused header for DataFormats packages #5

Following the issue #31505, the following headers are unused in several package:
 - EventFilter/DTRawToDigi/interface/DTROS25Data.h
 - EventFilter/ESRawToDigi/interface/ESCrcKchipFast.h
 - EventFilter/EcalRawToDigi/interface/ECALUnpackerException.h
 - EventFilter/EcalRawToDigi/interface/MyWatcher.h
 - EventFilter/RPCRawToDigi/interface/RPCRawSynchro.h
 - EventFilter/Utilities/interface/ModuleWeb.h
 - EventFilter/Utilities/interface/config_json.h
 - FWCore/ServiceRegistry/interface/TypeInfoHolder.h
 - GeneratorInterface/TauolaInterface/interface/TauolaWrapper.h
 - HeavyFlavorAnalysis/SpecificDecay/interface/BPHMuonChargeSelect.h
 - HeavyFlavorAnalysis/SpecificDecay/interface/BPHParticleChargeSelect.h
 - IOPool/Common/interface/CustomStreamer.h
 - JetMETCorrections/InterpolationTables/interface/BoxNDScanner.h
 - JetMETCorrections/InterpolationTables/interface/MultivariateFunctorScanner.h
 - JetMETCorrections/InterpolationTables/interface/rescanArray.h
 - JetMETCorrections/Type1MET/interface/CorrectedMETProducerT.h
 - L1Trigger/DTSectorCollector/interface/DTSCCand.h
 - L1Trigger/GlobalCaloTrigger/interface/L1GctSimpleJetFinder.h
 - L1Trigger/L1TCalorimeter/interface/classes.h
 - L1Trigger/L1TMuonOverlap/interface/OMTFConfigurator.h
 - L1Trigger/L1TMuonOverlap/interface/OMTFPatternMaker.h
 - L1Trigger/L1TNtuples/interface/L1AnalysisRecoTrackBase.h
 - L1TriggerOffline/L1Analyzer/interface/L1PromptAnalysis.h
 - MagneticField/Engine/interface/MagneticFieldHelpers.h
 - MagneticField/VolumeGeometry/interface/RectangularVolumeBounds.h
 - Mixing/Base/interface/FixedPUGenerator.h
 - Mixing/Base/interface/PUGenerator.h

#### PR validation:

Code compiles. Run local limited runTheMatrix.py tests.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Not a backport and no backport forseen.

### CODE REVIEW

**Comment by yuanchao** [comment]
@cmsbuild please test

**Comment by smorovic** [comment]
+1
ok to remove unused headers in EventFilter/Utilities

**Comment by makortel** [comment]
+core

**Comment by civanch** [comment]
+1

**Comment by agrohsje** [comment]
+generators

**Comment by jpata** [comment]
Pinging the relevant reco-area experts here. Jenkins tests show no difference, but are any of these headers used in non-standard ways, e.g. outside of workflows, in private scripts etc, as far as you know?  

@cms-sw/muon-dpg-l2
- EventFilter/DTRawToDigi/interface/DTROS25Data.h
- EventFilter/RPCRawToDigi/interface/RPCRawSynchro.h

@cms-sw/ecal-dpg-l2
- EventFilter/ESRawToDigi/interface/ESCrcKchipFast.h
- EventFilter/EcalRawToDigi/interface/ECALUnpackerException.h
- EventFilter/EcalRawToDigi/interface/MyWatcher.h

@camclean @alefisico
- JetMETCorrections/InterpolationTables/interface/BoxNDScanner.h
- JetMETCorrections/InterpolationTables/interface/MultivariateFunctorScanner.h
- JetMETCorrections/InterpolationTables/interface/rescanArray.h
- JetMETCorrections/Type1MET/interface/CorrectedMETProducerT.h

@namapane
- MagneticField/Engine/interface/MagneticFieldHelpers.h
- MagneticField/VolumeGeometry/interface/RectangularVolumeBounds.h

**Comment by thomreis** [comment]
For ECAL, I am not aware that these ECAL headers are used elsewhere.

**Comment by namapane** [comment]
The listed headers for MagneticField are indeed not used elsewhere, OK to remove.

**Comment by namapane** [comment]
adding @battibass for EventFilter/DTRawToDigi/interface/DTROS25Data.h 
I think that this one is indeed unused as there's another version of the same class in DataFormats/DTDigi/interface/DTControlData.h, which is the one actually used.

**Comment by battibass** [comment]
@namapane , thanks for having added me. I confirm what you say.
Up to a certain momenti in time, DT had some code performing data integrity monitoring without storing anything into the event, then we switched to the use of DataFormats. Therefore, IMHO, `EventFilter/DTRawToDigi/interface/DTROS25Data.h` is very likely just a relic of the old architecture.
I am not aware of any program running outside central workflows that relies on that file.

**Comment by mileva** [comment]
For RPC: 
I am not aware of any usage of EventFilter/RPCRawToDigi/interface/RPCRawSynchro.h 
We use similar class from the DataFormats.

**Comment by jpata** [comment]
Thank you all for responding above. Still need to confirm with JetMET:

pinging again @camclean @alefisico, please confirm if there's any non-CMSSW (outside a tested workflow) use of these headers in JME.
- JetMETCorrections/InterpolationTables/interface/BoxNDScanner.h
- JetMETCorrections/InterpolationTables/interface/MultivariateFunctorScanner.h
- JetMETCorrections/InterpolationTables/interface/rescanArray.h
- JetMETCorrections/Type1MET/interface/CorrectedMETProducerT.h

**Comment by camclean** [comment]
Checking with the current reco contacts. @laurenhay @michaelwassmer

**Comment by santocch** [comment]
+1

**Comment by camclean** [comment]
The first three JME header files should be fine to remove. We're checking on the fourth.

**Comment by michaelwassmer** [comment]
I had a look at the CorrectedMETProducer and I'd hesitate to remove it. It is the base class of e.g. 
JetMETCorrections/Type1MET/plugins/CorrectedPFMETProducer.cc
and if you search for this producer alone one finds some instances in HLTTrigger, see 
https://cmssdt.cern.ch/lxr/search?%21v=CMSSW_12_1_X_2021-09-13-2300&_filestring=&_string=CorrectedPFMETProducer
I'm not sure if these are important or not. Also the CorrectedPFMETProducer is imported e.g. by
JetMETCorrections/Type1MET/python/correctedMet_cff.py
which is imported on other instances, see
https://cmssdt.cern.ch/lxr/search?%21v=CMSSW_12_1_X_2021-09-13-2300&_filestring=&_string=correctedMET_cff
I think the probability that the correction sequences for MET are used in some production sequences/steps are not that low.
In summary, I guess I would keep that for now, maybe @lathomas can comment on this with more expertise.

**Comment by jpata** [comment]
Thanks JME folks for the follow-up. So it looks like all these headers can be removed indeed.

**Comment by jpata** [comment]
+reconstruction
- for b09e121
- technical, removes unused headers
- in the reco area, we explicitly checked the headers of the various areas for nonstandard/out-of-workflows use: https://github.com/cms-sw/cmssw/pull/35105#issuecomment-913655306
- no reco differences expected/observed

**Comment by qliphy** [comment]
please test

**Comment by qliphy** [comment]
+1

**Comment by qliphy** [comment]
please test

Several files are adding back: https://github.com/jrotter2/cmssw/commit/ceb35356874fabc3041bab8c0cd8bbf8346c8383

@cms-sw/l1-l2  Please have a look.

**Comment by jpata** [comment]
I'm confused about the latest update in ceb3535. Was e.g. CorrectedMETProducerT.h now added back? Can you please check that the description is up to date?

**Comment by qliphy** [comment]
> I'm confused about the latest update in [ceb3535](https://github.com/cms-sw/cmssw/commit/ceb35356874fabc3041bab8c0cd8bbf8346c8383). Was e.g. CorrectedMETProducerT.h now added back? Can you please check that the description is up to date?

@yuanchao FYI

**Comment by jpata** [comment]
OK, as far as I can tell, nothing was added back. It's simply that a commit in a private fork of CMSSW referenced this issue, which caused it to show up here (a retest was consequently not really necessary I think).

Looking at the commit history of this PR (https://github.com/cms-sw/cmssw/pull/35105/commits), there is just one commit, the removal of all the files listed. So no complaints from reco.

**Comment by qliphy** [comment]
+1
